google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/models/ChangeStreamMutationTest.java

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.models;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.bigtable.v2.MutateRowRequest;
+import com.google.bigtable.v2.MutateRowsRequest;
+import com.google.bigtable.v2.ReadChangeStreamResponse;
+import com.google.cloud.bigtable.data.v2.internal.NameUtil;
+import com.google.cloud.bigtable.data.v2.internal.RequestContext;
+import com.google.common.primitives.Longs;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Timestamp;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ChangeStreamMutationTest {
+  private static final String PROJECT_ID = "fake-project";
+  private static final String INSTANCE_ID = "fake-instance";
+  private static final String TABLE_ID = "fake-table";
+  private static final String APP_PROFILE_ID = "fake-profile";
+  private static final RequestContext REQUEST_CONTEXT =
+      RequestContext.create(PROJECT_ID, INSTANCE_ID, APP_PROFILE_ID);
+
+  @Rule public ExpectedException expect = ExpectedException.none();
+
+  @Test
+  public void userInitiatedMutationTest() throws IOException, ClassNotFoundException {
+    // Create a user initiated logical mutation.
+    Timestamp fakeCommitTimestamp = Timestamp.newBuilder().setSeconds(1000).build();
+    Timestamp fakeLowWatermark = Timestamp.newBuilder().setSeconds(2000).build();
+    ChangeStreamMutation changeStreamMutation =
+        ChangeStreamMutation.create(
```

Comment:
I wonder if you could create a builder for ChangeStreamMutation, and check if tokens and low watermarks are set when you call build().

So something like:
ChangeStreamMutation.Builder = ChangeStreamMutation.createUserMutation(...)
        .deleteFamaly(..).setCell(..).setToken(..).setLowWatermark(..);

builder.build() will throw exception if toke nand watermakr are not set.

Reply:
Example: https://github.com/googleapis/java-bigtable/blob/8b120de58f0dfba3573ab696fb0e5375e917a00e/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java#L836-L837